File: python/selenium/configs.py
import os

# Accessing variables.
kobiton_username = os.getenv('KOBITON_USERNAME')
kobiton_access_key = os.getenv('KOBITON_ACCESS_KEY')
device_name = os.getenv('DEVICE_NAME')
kobiton_app = os.getenv('KOBITON_APP')

print(os.environ)

kobitonServerUrl = 'https://{0}:{1}@api.kobiton.com/wd/hub'.format(kobiton_username, kobiton_access_key)
print(kobitonServerUrl)
# 100 seconds
session_timeout = 120

desired_caps_android_web = {
  'sessionName': '[Python] Android web',
  'sessionDescription': 'This is an example for Android web testing',
  'deviceOrientation':  'portrait',
  'browserName':        'chrome',
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         'Galaxy S5',
  'platformName':       'Android',
  'newCommandTimeout':  120
}

desired_caps_android_app = {
  'sessionName': '[Python] Android app',
  'sessionDescription': 'This is an example for Android app testing',
  'deviceOrientation':  'portrait',
  'app':                kobiton_app or 'https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.apk',
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         device_name or '*',
  'platformName':       'Android',
  'newCommandTimeout':  120
}

desired_caps_ios_web = {
  'sessionName': '[Python] iOS web',
  'sessionDescription': 'This is an example for iOS web testing',
  'deviceOrientation':  'portrait',
  'browserName':        'safari', 
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         'iPhone 7',
  'platformName':       'iOS',
  'newCommandTimeout':  120
}

desired_caps_ios_app = {
  'sessionName': '[Python] iOS app',
  'sessionDescription': 'This is an example for iOS app testing',
  'deviceOrientation':  'portrait',
  'app':                'https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.ipa',
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         'iPhone 7',
  'platformName':       'iOS',
  'newCommandTimeout':  120
}

File: java/Cucumber/src/test/java/configs/Configs.java
package configs;

import java.net.MalformedURLException;
import java.net.URL;
import org.openqa.selenium.remote.DesiredCapabilities;

public class Configs {

    static String USERNAME = "";
    static String ACCESS_KEY = "";
    public static final URL kobitonServerUrl(){
        try {
            return new URL("https://khanhdo:9e92684c-68a1-4882-a499-8894f8715bd0@api-test.kobiton.com/wd/hub");
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static final DesiredCapabilities desiredCapabilitiesAndroidWeb(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "Android Web xxx");
        capabilities.setCapability("sessionDescription", "This is an example for Android Web testing xxx");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("browserName", "chrome");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "*");
        capabilities.setCapability("platformName", "Android");
        return capabilities;
    }

    public static final DesiredCapabilities desiredCapabilitiesAndroidApp(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "Android app xxx");
        capabilities.setCapability("sessionDescription", "Automation test android app session xxx");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.apk");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "*");
        capabilities.setCapability("platformName", "Android");
        return capabilities;
    }

    public static final DesiredCapabilities desiredCapabilitiesiOSWeb(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "iOS Web xxx");
        capabilities.setCapability("sessionDescription", "This is an example for iOS Web testing xxx");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("browserName", "safari");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "*");
        capabilities.setCapability("platformName", "iOS");
        return capabilities;
    }

    public static final DesiredCapabilities desiredCapabilitiesiOSApp(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "iOS app xxx");
        capabilities.setCapability("sessionDescription", "This is an example for iOS App testing xxx");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.ipa");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "*");
        capabilities.setCapability("platformName", "iOS");
        return capabilities;
    }
}